### ISSUE TITLE
999-999 Value Error while reading HURDAT

### ISSUE THREAD

**Issue by Justin5999** [tropycal/tropycal#179]
_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data '999-999' does not match format '%Y%m%d%H%M'

Error listed above is not allowing tropycal to read HURDAT2 data. Is there a fix for this bug?

Thanks,
Justin

**Comment by bmaschino** [comment]
Getting this error too when reading HURDAT2 for the North Atlantic but not the East Pacific. I think something in HURDAT2 is out of alignment and making the strptime function get 999-999 as an argument. It looks like at the start of the 1967 hurricane season, there is a comma after the last column which doesn't exist prior to 1967. The East Pacific dataset does not include this trailing comma


19661126, 0000,  , TD, 35.0N,  44.2W,  30, -999, -999, -999, -999, -999, -999, -999, -999, -999, -999, -999, -999, -999, -999
AL011967,            UNNAMED,      9,
19670610, 1200,  , TD, 17.0N,  85.0W,  25, 1006, -999, -999, -999, -999, -999, -999, -999, -999, -999, -999, -999, -999,
19670610, 1800,  , TD, 17.3N,  85.0W,  25, -999, -999, -999, -999, -999, -999, -999, -999, -999, -999, -999, -999, -999,

**Comment by Justin5999** [comment]
Agreed. I found a work around if you replace the HURDAT2 website with a local file of the database is tends to work. 

This is good for work but will not be helpful for recon or real-time analysis

**Comment by tomerburg** [comment]
Thanks for opening this issue! Yes, this is an issue with the latest revised version of HRD's HURDATv2 file as @bmaschino pointed out. The next Tropycal release (currently in the develop branch) will use the NHC version of HURDATv2, but for now you can also avert this by replacing the ``atlantic_url`` argument with the NHC HURDATv2 URL:

```
url = "https://www.nhc.noaa.gov/data/hurdat/hurdat2-1851-2022-042723.txt"
basin = tracks.TrackDataset(basin='north_atlantic', atlantic_url=url)
```

This should allow your code to work with recon & realtime analysis normally -- please let us know if you encounter any unexpected issues with this approach!

**Comment by tomerburg** [comment]
This bug was fixed with the release of Tropycal v1.0, so I'm closing this issue. Thanks again for reporting this and please let us know if you encounter any other problems!
